lib/buckets/buckets.js

```diff
@@ -142,6 +146,8 @@ function BucketObject(req) {
     self.bucket_name = req.params.bucket_name;
     if (req.params.object_name) {
         self.name = req.params.object_name;
+        self.name_hash =
+            crypto.createHash('md5').update(self.name).digest('hex');
```

Comment:
You decided on md5 for speed?

Reply:
Speed and it didn't require any other dependencies.